backend/editor/entries.py

```diff
@@ -0,0 +1,60 @@
+from dependencies import *
+
+def initialize_db():
+    global driver, session
+    uri = settings.uri
+    driver = GraphDatabase.driver(uri)
+    session = driver.session()
+
+def shutdown_db():
+    session.close()
+    driver.close()
+
+def get_all_nodes(label):
+    if label == "":
+        query = f"""
+            MATCH (n) RETURN n
+        """
+    else:
+        query = f"""
+            MATCH (n:{label}) RETURN n
+        """
+    result = session.run(query)
+    return result
+
+def get_nodes(label, entry):
+    query = f"""
+        MATCH (n:{label}) WHERE n.id = {entry} 
+        RETURN n
+    """
+    result = session.run(query)
+    return result
+
+def get_marginals(id):
+    query = f"""
+        MATCH (n:TEXT) WHERE n.id = {id}
+        RETURN n
+    """
+    result = session.run(query)
+    return result
+
+def update_nodes(label, entry, incomingData):
+    query = f"""MATCH (n:{label}) WHERE n.id = $id"""
+
+    for (key,value) in incomingData.items():
+        query += f"""\nSET n.{key} = {value}\n"""
+
+    query += f"""RETURN n"""
+
+    result = session.run(query, {"id": entry})

```

Comment:
It's a bit silly, because we already inject keys, but non the less, I will use parameters to inject values.

To avoid [injection](https://developer.mozilla.org/en-US/docs/Glossary/SQL_Injection) problems, we may want to check keys only use word.

Last: when you build a query like this, the best approach is to collect lines in a list and join them at the end (faster, cleaner)

So I propose:

```suggestion
    # sanity check keys
    for key in incomingData.keys():
        if not re.match(r"^\w+$", key) or key == "id":
            raise ValueError("Invalid key: %s", key)
    # build query
    query = [f"""MATCH (n:{label}) WHERE n.id = $id"""]
    for key in incomingData.keys():
        query.append(f"""\nSET n.{key} = ${key}\n""")
    query.append(f"""RETURN n""")
    params = dict(incomingData, id=entry)
    result = session.run(" ".join(query), params)
```